src/ui/public/index_patterns/index_patterns.js

```diff
@@ -41,6 +41,7 @@ export function IndexPatternsProvider(Notifier, Private) {
   self.cache = patternCache;
   self.getIds = getProvider('id');
   self.getTitles = getProvider('attributes.title');
+  self.getIndexPatterns = getProvider();
```

Comment:
Actually now that I think about it a bit more, I don't think the `indexPattern` suffix is necessary.  The other functions aren't `indexPatternIntervals`, `indexPatternCache`, `indexPatternFieldsFetcher`, etc.  Adding that might be taking it to far, and if you aren't going to add it to all, why add it to those two?